File: tests/fixtures/SsrfToctouSafe.java
import java.net.URL;
import java.nio.file.Files;
import java.nio.file.NoSuchFileException;
import java.nio.file.Path;

final class SsrfToctouSafe {
    Object fetchHealthcheck() throws Exception {
        return new URL("https://status.example.com/health").openConnection();
    }

    String readFile(Path path) throws Exception {
        try {
            return Files.readString(path);
        } catch (NoSuchFileException exception) {
            return "";
        }
    }
}

File: tests/fixtures/ssrf_toctou_safe.go
package fixtures

import (
	"net/http"
	"os"
)

func fetchHealthcheck() (*http.Response, error) {
	return http.Get("https://status.example.com/health")
}

func readFile(path string) ([]byte, error) {
	return os.ReadFile(path)
}
